### PR TITLE
Update README.md [Minor changes on developer's guide]

### PR DESCRIPTION
**Pull request by davidchoy98**
Modify the links to the specific documentation on developer's guide.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,7 +59,12 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
 Before you contribute, please read through the contributing and developer guides to get an idea of what kinds of pull requests we will or won't accept.
 
 * [Contributing Guide](documentation/wiki/Contributing-Code.md)
-* [Developer Guide](documentation/wiki/Building-Testing-and-Debugging.md)
+
+* **Developer Guide on:**
+   - [.Net Core](documentation/wiki/Building-Testing-and-Debugging-on-.Net-Core-MSBuild.md)
+   - [Full Framework](documentation/wiki/Building-Testing-and-Debugging-on-Full-Framework-MSBuild.md)
+   - [Mono](documentation/wiki/Building-Testing-and-Debugging-on-Mono-MSBuild.md)
```
[the remark this answers is not in the document]
Comment (radical) [reply README.md]: Umm.. it needs some changes like:

- netcore 1.x is not used now, so IIUC, we don't need the special openssl dance anymore.
- Also, the second branch name in `You can clone that and use the xplat-master branch or mono-2017-10 for the next release branch` needs to be changed to `mono-2018-04`.

**Comment by davidchoy98** [comment]
@radical please review on the latest update.

File: documentation/wiki/Building-Testing-and-Debugging-on-Mono-MSBuild.md [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
 
 # Unix #
 
 **Required packages for OSX & Ubuntu**
 
-MSBuild currently builds with a prerelease version of .NET Core 1.0. It requires the [.NET Core prerequisites](https://github.com/dotnet/core/blob/master/Documentation/prereqs.md), which you can acquire manually or easily get by [installing the .NET Core SDK](https://dot.net/core).
-
-* *OpenSSL*: MSBuild uses the .Net CLI to download Nuget packages during its build process. The CLI requires a recent OpenSSL library available in `/usr/lib`. This can be downloaded using [brew](http://brew.sh/) on OS X (`brew install openssl`) and apt-get (`apt-get install openssl`) on Ubuntu, or [building from source](https://wiki.openssl.org/index.php/Compilation_and_Installation#Mac). If you use a different package manager and see an error that says `Unable to load DLL 'System.Security.Cryptography.Native'`, `dotnet` may be looking in the wrong place for the library.
+MSBuild currently builds with .NET Core 2.1. It requires the [.NET Core prerequisites](https://github.com/dotnet/core/blob/master/Documentation/prereqs.md), which you can acquire manually or easily get by [installing the .NET Core SDK](https://dot.net/core).
```
Comment (radical) [inline documentation/wiki/Building-Testing-and-Debugging-on-Mono-MSBuild.md]: For the mono fork, you just need a recent version of mono installed.

Comment (davidchoy98) [reply documentation/wiki/Building-Testing-and-Debugging-on-Mono-MSBuild.md]: So IIUC, there's no need to have the .Net Core Prerequisites or the SDK?

Comment (radical) [reply documentation/wiki/Building-Testing-and-Debugging-on-Mono-MSBuild.md]: I have not tested this on a fresh machine, but yeah. I can build without having the dotnet sdk installed. We use a bootstrap mono+msbuild.

Comment (davidchoy98) [reply documentation/wiki/Building-Testing-and-Debugging-on-Mono-MSBuild.md]: @radical how about this '[libunwind](http://www.nongnu.org/libunwind/index.html) is required by .NET Core. Install it using `sudo apt-get install libunwind8`' ?

Comment (radical) [reply documentation/wiki/Building-Testing-and-Debugging-on-Mono-MSBuild.md]: It shouldn't be, but I'm not sure. @directhex would know better.

Comment (directhex) [reply documentation/wiki/Building-Testing-and-Debugging-on-Mono-MSBuild.md]: It looks like .NET Core (and its dependencies) are no longer needed.

I'd suggest a note regarding bypassing the Nerdbank.GitVersioning mess (it only works on old distributions without OpenSSL 1.1), though.

Comment (rainersigwald) [reply documentation/wiki/Building-Testing-and-Debugging-on-Mono-MSBuild.md]: Nerdbank.GitVersioning should be fixed now with #3657, fortunately.

**Pull request merged**